feat: add ITS contract addresses for mainnet and testnet

Olanetsoft opened the pull request.

StephenFluin reviewed (changes requested):
Token managers won't have an address that is meaningful for developers. How would this be used? I recommend we remove everything except the service. I also would love for you to link to a block explorer for each of these.

Olanetsoft commented:
> Token managers won't have an address that is meaningful for developers. How would this be used? I recommend we remove everything except the service. I also would love for you to link to a block explorer for each of these.

I added the Token Manager address because it's useful for developers. It's useful for developers when linking deployed tokens on multiple chains into Interchain Token to encode params. However, I have removed it in case it seems irrelevant. Additionally, I have updated the PR by adding a link to the block explorer.

StephenFluin commented:
Let's make sure I understand, every new token that gets deployed gets its
own token manager with its own address. So what is this token manager?

Stephen Fluin

On Fri, Jan 5, 2024, 5:03 PM Idris Olubisi ***@***.***> wrote:

> Token managers won't have an address that is meaningful for developers.
> How would this be used? I recommend we remove everything except the
> service. I also would love for you to link to a block explorer for each of
> these.
>
> I added the Token Manager address because it's useful for developers. It's
> useful for developers when linking deployed tokens on multiple chains into
> Interchain Token to encode params. However, I have removed it in case it
> seems irrelevant. Additionally, I have updated the PR by adding a link to
> the block explorer.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/axelarnetwork/axelar-docs/pull/697#issuecomment-1879360613>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABIJQAJQMCSJ7AP73DVMDDYNCBCNAVCNFSM6AAAAABBPBBK6WVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQNZZGM3DANRRGM>
> .
> You are receiving this because your review was requested.Message ID:
> ***@***.***>
>

Olanetsoft commented:
> Let's make sure I understand, every new token that gets deployed gets its own token manager with its own address. So what is this token manager? Stephen Fluin
> […](#)
> On Fri, Jan 5, 2024, 5:03 PM Idris Olubisi ***@***.***> wrote: Token managers won't have an address that is meaningful for developers. How would this be used? I recommend we remove everything except the service. I also would love for you to link to a block explorer for each of these. I added the Token Manager address because it's useful for developers. It's useful for developers when linking deployed tokens on multiple chains into Interchain Token to encode params. However, I have removed it in case it seems irrelevant. Additionally, I have updated the PR by adding a link to the block explorer. — Reply to this email directly, view it on GitHub <[#697 (comment)](https://github.com/axelarnetwork/axelar-docs/pull/697#issuecomment-1879360613)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AABIJQAJQMCSJ7AP73DVMDDYNCBCNAVCNFSM6AAAAABBPBBK6WVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQNZZGM3DANRRGM> . You are receiving this because your review was requested.Message ID: ***@***.***>

Yes, correct. Each new ITS token receives its own token manager with a unique address. However, when linking deployed tokens across multiple chains through Interchain Token, deploying a token manager becomes necessary. This process requires calling the `params` method on the Token Manager contract, specifically this one: [0xC1B09c9c16117417A1B414A52Dd92CF1f634e786](https://etherscan.io/address/0xC1B09c9c16117417A1B414A52Dd92CF1f634e786). These parameters are part of the requirements for deploying a TM, as referenced [here](https://github.com/axelarnetwork/interchain-token-service/blob/9223108211b977d9138fdd67d5b4a641fc35c18c/contracts/InterchainTokenService.sol#L280). That's why I think it would be beneficial to have this TM address visible in our documentation as well. @StephenFluin
